gapis/gfxapi/vulkan/dependency_graph.go

```diff
@@ -0,0 +1,1733 @@
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+	case *VkCmdSetLineWidth:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+	case *RecreateCmdSetLineWidth:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+	case *VkCmdSetScissor:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+	case *RecreateCmdSetScissor:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+	case *VkCmdSetViewport:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+	case *RecreateCmdSetViewport:
+		addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+		addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+
+	case *VkCmdBindDescriptorSets:
+		descriptorSetCount := typedAtom.DescriptorSetCount
+		descriptorSets := typedAtom.PDescriptorSets.Slice(0, uint64(descriptorSetCount), s)
+		for i := uint32(0); i < descriptorSetCount; i++ {
+			descriptorSet := descriptorSets.Index(uint64(i), s).Read(ctx, a, s, nil)
+			addRead(&b, g, vulkanStateKey(descriptorSet))
+			for _, descBinding := range GetState(s).DescriptorSets.Get(descriptorSet).Bindings {
+				for _, bufferInfo := range descBinding.BufferBinding {
+					buf := bufferInfo.Buffer
+					addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+					addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+					recordCommand(typedAtom.CommandBuffer, func(b *AtomBehaviour) {
+						// Descriptors might be modified
+						addModify(b, g, vulkanStateKey(buf))
+						// Advance the read/modify behavior of the descriptors from
+						// draw and dispatch calls to here. Details in the handling
+						// of vkCmdDispatch and vkCmdDraw.
+						modifyMemoryBindingsData(b, getOverlappedBindingsForBuffer(buf))
+					})
+				}
+				for _, imageInfo := range descBinding.ImageBinding {
+					view := imageInfo.ImageView
+					addRead(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).handle)
+					addModify(&b, g, g.getOrCreateCommandBuffer(typedAtom.CommandBuffer).records)
+					recordCommand(typedAtom.CommandBuffer, func(b *AtomBehaviour) {
+						addRead(b, g, vulkanStateKey(view))
+						img := GetState(s).ImageViews.Get(view).Image.VulkanHandle
+						// Advance the read/modify behavior of the descriptors from
+						// draw and dispatch calls to here. Details in the
```

Comment:
I think this is missing the special logic for writing over the end of your descriptor arrays.